src/client/context.cc

```diff
@@ -359,6 +358,29 @@ void ClientContext::UpdateClientTunnel(
       rebind =
           (client_tunnel->GetEndpoint()
            != boost::asio::ip::tcp::endpoint(next_addr, tunnel.port));
+    if (!rebind)  // check for change in destination
+      {
+        LOG(trace)
+            << "ClientContext: update checking for change in destination";
+        rebind = (client_tunnel->GetTunnelAttributes().dest != tunnel.dest)
+                 || (client_tunnel->GetTunnelAttributes().dest_port
+                     != tunnel.dest_port);
+      }
+    if (!rebind)  // check for change in keys
+      {
+        LOG(trace) << "ClientContext: update checking for change in keys";
```

Comment:
`debug` please. I think we should save `trace` for encoded binary output and finer debugging.